### PR TITLE
feat(segment): add `bun`

### PR DESCRIPTION
**Pull request by Joxtacy**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

### Description

Adds a segment for [`bun`](https://bun.sh/).

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by Joxtacy** [comment]
I tried to add 🥟 as the emoji using the code point `\0001f95f`, but it doesn't render correctly in the docs.
3145da00de787b932b7ecf6f175e97a7


It works as expected in my terminal (alacritty) when I add it in my template `template = "[\U0001f95f {{ .Full }}]"` though.
Screenshot 2024-06-09 at 11 51 00


Any input on how I should proceed here?

**Review by JanDeDobbeleer — APPROVED** [review]

File: src/segments/bun.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+func (b *Bun) Template() string {
+	return languageTemplate
+}
+
+func (b *Bun) Init(props properties.Properties, env platform.Environment) {
+	b.language = language{
+		env:        env,
+		props:      props,
+		extensions: []string{"*.js", "*.ts", "bun.lockb"},
```
Comment (JanDeDobbeleer) [inline src/segments/bun.go]: do we want to use the same extensions as the node segment? Or only trigger on `bun.lock`?

**Comment by JanDeDobbeleer** [comment]
@Joxtacy maybe the docs Nerd Font isn't V3 yet?

File: src/segments/bun.go (hunk 1 above)
Comment (Joxtacy) [reply src/segments/bun.go]: I was thinking about this as well. When testing things out, I noticed myself setting `extensions = ["bun.lockb"]` anyway. 
I more or less just copied what we have for `Deno`. Technically you could use the bun cli outside of a bun project.
I don't really have any strong preferences here. Personally I think just having the `bun.lockb` is fine.

**Comment by Joxtacy** [comment]
> @Joxtacy maybe the docs Nerd Font isn't V3 yet?

It seems like it's the `JSON.stringify`, `YAML.stringify`, `TOML.stringify` functions that aren't replacing the code point correctly. I could create a makeshift replacer function for both JSON and YAML to manually call `String.fromCodePoint("0x0001f95f")` and that then renders the emoji correctly.

Not sure what to do about that. Maybe I can just find another icon that plays more nicely. 🙃

File: src/segments/bun.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/segments/bun.go]: I would indeed keep `bun.lockb` only as the default, people can override as they see fit.

**Comment by JanDeDobbeleer** [comment]
@Joxtacy what if you paste in the actual emoji rather than the code point?

**Comment by Joxtacy** [comment]
> @Joxtacy what if you paste in the actual emoji rather than the code point?

That works just fine. I thought you wanted to have the code points, but maybe I misunderstood the contribution docs.

**Comment by JanDeDobbeleer** [comment]
> I thought you wanted to have the code points

@Joxtacy back in the day, check, but now as they're being rendered, this isn't required anymore.

File: src/segments/bun.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/segments/bun.go]: @Joxtacy if you adjust this part, we're ready to merge 👍🏻

Comment (Joxtacy) [reply src/segments/bun.go]: Done! :) If you'd just like to double check to make sure I didn't miss anything, I'd be happy as well.

**Review by JanDeDobbeleer — APPROVED** [review]

File: src/segments/bun.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/segments/bun.go]: already did, and if not, that's just a small PR away 😄

**Comment by Joxtacy** [comment]
I forgot to update my test... 🙃

**Comment by JanDeDobbeleer** [comment]
@all-contributors please add @Joxtacy for code,design,doc

This is pure magic 🪄! @holopin-bot @Joxtacy wizard

**Pull request merged**